### DISCUSSION TITLE
ReactDOM.render - What Is It Doing?

### DISCUSSION THREAD

**Discussion by coltmay** [nss-day-cohort-47/C47-Discussions#81] [Help Me] [answered]
So I'm trying to go line-by-line to determine what isn't clear to me, and guess what?  I'm unsure about something at the very beginning of the project.  :)  I'm going to explain what I _think_ is happening, and can somebody then correct me?

In the following code...
```javascript
// ReactDOM.render takes the contents being passed in and renders it to the DOM...
ReactDOM.render(
// React.StrictMode...I'm not sure?  Is it used for debugging and testing?
  <React.StrictMode>
// <Router> confuses me.  I get that it's just something that must always be implemented in .render(right?), but what does it do?  Does it allow <Route> to function in the rest of our code?
      <Router>
// <Kennel> is the heart of the web-app, it is essentially the "root" of all of the dynamic things that we love about react and JSX elements.
          <Kennel />
      </Router>
  </React.StrictMode>,
// At least this is familiar.  Judging by the comma on the line above, I assume this is an argument that tells .render where to place the rendered content?  In this case, injecting it where the "root" id is on the HTML file.
  document.getElementById("root")
);
```

**Comment by ardisrtilson** [comment, marked as answer]
Router is among the set of tools we use to allow for dynamic routing in React Apps. We're enclosing the main Kennel app within the Router function call in the JSX so these tools will be available in the components below. 

<React.StrictMode> tells React to throw some errors that it wouldn't otherwise throw for debugging and learning purposes. 

ReactDOM.render() is the syntax for making a render call for the entire react app. Everything that happens in the React App begins with that main router call. 

This particular file is kind of heavy and not the main idea of what we're teaching now, but there's some brief answers. If you're looking for ins and outs, I'd look at the documentation for ReactDOM, StrictMode, and Router, but if it's too dense at the moment I wouldn't sweat it, and trust it'll become clearer over time.

**Reply by coltmay** [reply]
Thanks Red!  That clears things up a lot.  I just like having as much understanding as possible.
